cmd/soroban-cli/src/commands/init.rs

```diff
@@ -0,0 +1,329 @@
+use std::fs::read_to_string;
+use std::path::Path;
+use std::{fs, io};
+
+use clap::Parser;
+use std::num::NonZeroU32;
+use std::sync::atomic::AtomicBool;
+use toml_edit::{Document, Formatted, InlineTable, TomlError, Value};
+
+#[derive(Clone, Debug, PartialEq, clap::ValueEnum)]
+pub enum ExampleContract {
+    Account,
+    Alloc,
+    AtomicMultiswap,
+    AtomicSwap,
+    Auth,
+    CrossContract,
+    CustomTypes,
+    DeepContractAuth,
+    Deployer,
+    Errors,
+    Events,
+    Fuzzing,
+    HelloWorld,
+    Increment,
+    LiquidityPool,
+    Logging,
+    SimpleAccount,
+    SingleOffer,
+    Timelock,
+    Token,
+    UpgradeableContract,
+    None,
+}
+
+impl ToString for ExampleContract {
+    fn to_string(&self) -> String {
+        match self {
+            ExampleContract::Account => String::from("account"),
+            ExampleContract::Alloc => String::from("alloc"),
+            ExampleContract::AtomicMultiswap => String::from("atomic_multiswap"),
+            ExampleContract::AtomicSwap => String::from("atomic_swap"),
+            ExampleContract::Auth => String::from("auth"),
+            ExampleContract::CrossContract => String::from("cross_contract"),
+            ExampleContract::CustomTypes => String::from("custom_types"),
+            ExampleContract::DeepContractAuth => String::from("deep_contract_auth"),
+            ExampleContract::Deployer => String::from("deployer"),
+            ExampleContract::Errors => String::from("errors"),
+            ExampleContract::Events => String::from("events"),
+            ExampleContract::Fuzzing => String::from("fuzzing"),
+            ExampleContract::HelloWorld => String::from("hello_world"),
+            ExampleContract::Increment => String::from("increment"),
+            ExampleContract::LiquidityPool => String::from("liquidity_pool"),
+            ExampleContract::Logging => String::from("logging"),
+            ExampleContract::SimpleAccount => String::from("simple_account"),
+            ExampleContract::SingleOffer => String::from("single_offer"),
+            ExampleContract::Timelock => String::from("timelock"),
+            ExampleContract::Token => String::from("token"),
+            ExampleContract::UpgradeableContract => String::from("upgradeable_contract"),
+            ExampleContract::None => String::from("none"),
+        }
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    pub project_path: String,
+
+    /// optional flag to specify soroban example contracts to include
+    #[arg(short, long, num_args = 1..=20, default_value = "none")]
```

Comment:
Is it worth including a set of common contracts by default (e.g. `hello-world`) rather than have the default include no example contracts?

Reply:
Instead of hello world, which tbh is a bit of an odd contract (my fault, I wrote it), I think we should include something like the increment / storage example. We could chop it down a bit, but Im thinking something simple but functional that anyone can understand like adding two numbers of incrementing a stored number. And ideally we also include a test in a test file, and a fuzz test in a fuzz setup. Thoughts?

Reply:
Also I realize fuzzing is an advanced thing, but I think we've gotta push hard on fuzzing to make it adopted.

Reply:
Making this as resolved - I've added the increment contract to be included by default. I'll plan to look at the fuzz setup piece in a follow-up PR.